### PR TITLE
memory leak fix

### PR DESCRIPTION
**Pull request by AShivangi**

### CODE REVIEW

File: src/runtime_src/core/tools/common/XBUtilities.cpp [hunk 1]
```diff
@@ -445,11 +445,11 @@ XBUtilities::get_axlf_section(const std::string& filename, axlf_section_kind kin
...
     throw std::runtime_error("Section not found");
 
-  auto buf = new char[section->m_sectionSize];
+  std::unique_ptr<char> buf(new char[section->m_sectionSize]);
   in.seekg(section->m_sectionOffset);
-  in.read(buf, section->m_sectionSize);
+  in.read(buf.get(), section->m_sectionSize);
 
-  return std::make_pair(buf, section->m_sectionSize);
+  return std::make_pair(buf.get(), section->m_sectionSize);
```
Comment (stsoe) [inline src/runtime_src/core/tools/common/XBUtilities.cpp:452]: No that wont work.  The unique_ptr goes out of scope at lone 453 and the returned buffer has been deleted.  You will have to pass ownership to the caller via std::unique_ptr in the returned pair.

File: src/runtime_src/core/tools/common/XBUtilities.cpp [hunk 2]
```diff
@@ -413,7 +413,7 @@ XBUtilities::report_available_devices()
   std::cout << std::endl;
 }
 
-std::pair<const char*, size_t>
+std::pair<std::vector<char>, size_t>
```
Comment (stsoe) [inline src/runtime_src/core/tools/common/XBUtilities.cpp:416]: just return std::vector.  size of vector is same.

Comment (AShivangi) [reply src/runtime_src/core/tools/common/XBUtilities.cpp:416]: Oh yes, good point ;)

**Comment by stsoe** [comment]
retest this please

**Pull request merged**